src/Uno.Foundation/Diagnostics/DiagnosticViewRegistry.cs

```diff
@@ -10,14 +10,14 @@ namespace Uno.Diagnostics.UI;
 /// </summary>
 internal static class DiagnosticViewRegistry
 {
-	internal static EventHandler<ImmutableList<DiagnosticViewRegistration>>? Added;
+	internal static EventHandler<IImmutableList<DiagnosticViewRegistration>>? Added;
 
-	private static ImmutableList<DiagnosticViewRegistration> _registrations = ImmutableList<DiagnosticViewRegistration>.Empty;
+	private static ImmutableArray<DiagnosticViewRegistration> _registrations = ImmutableArray<DiagnosticViewRegistration>.Empty;
 
 	/// <summary>
 	/// Gets the list of registered diagnostic providers.
 	/// </summary>
-	internal static ImmutableList<DiagnosticViewRegistration> Registrations => _registrations;
+	internal static IImmutableList<DiagnosticViewRegistration> Registrations => _registrations;
```

Comment:
Note: I think this will box. Not sure if it's a concern though. But wanted to note it.